### DISCUSSION TITLE
eCal lightweight version for Docker

### DISCUSSION THREAD

**Discussion by charlie34** [eclipse-ecal/ecal#1001] [General]
Hi at all,
I need a lightweight to implement it in a docker, since when I install this version
the docker takes 800MB, which is not admitted.
Could be it possible? Or with this version, can I install some parts from this project and it executes?
Thanks in advanced
Carlos

**Comment by FlorianReimold** [comment]
Hi Charlie,

For simplicity reasons we currently don't publish precompiled smaller debian packages. This is kind of on our roadmap, especially for the PPA, but it takes effort do do that and we just haven't done it, yet.
So when installing eCAL from our PPA in a docker container, you will also install all GUI apps, Qt Depdencies etc., which will blow up your container.

Currently, you will have to compile eCAL yourself and just turn on those parts of eCAL, that you need (maybe just the core runtime? Python? CLI Apps? You decide!)

Here you can find a guide on how to compile eCAL. Remember to check out the correct git tag, if you want to work with a released version of eCAL instead of the master: https://eclipse-ecal.github.io/ecal/development/building_ecal_from_source.html#fa-ubuntu-building-ecal-on-ubuntu
Here you can find the CMake Options we use for compiling eCAL. The more parts you turn off, the smaller the binary and the dependency list will get: https://github.com/eclipse-ecal/ecal/blob/a57825bd8b6b2c03c55d72333aad0866cd1c7849/.github/workflows/build-ubuntu-22.yml#L60-L97
